### ISSUE TITLE
support Tomcat 10.1.x for REST API

### ISSUE THREAD

**Issue by vladak** [oracle/opengrok#4075]
As discussed in https://github.com/oracle/opengrok/discussions/4070 , API search request in Tomcat 10.1.0 ends with:
```
org.glassfish.jersey.server.ContainerException: java.lang.NoSuchMethodError: 'void jakarta.servlet.http.HttpServletResponse.setStatus(int, java.lang.String)'
```

**Comment by vladak** [comment]
Comparing `HttpServletResponse` documentation for Tomcat [10.0](https://tomcat.apache.org/tomcat-10.0-doc/servletapi/jakarta/servlet/http/HttpServletResponse.html) and [10.1](https://tomcat.apache.org/tomcat-10.1-doc/servletapi/jakarta/servlet/http/HttpServletResponse.html) reveals that the `setStatus(int, java.lang.String)` is indeed gone in 10.1. In 10.0 it was marked as deprecated. [setStatus](https://tomcat.apache.org/tomcat-10.0-doc/servletapi/jakarta/servlet/http/HttpServletResponse.html#setStatus(int))​(int sc) remains to be used. Looking into OpenGrok code, I do not see the method being used other than in the `CookieFilterTest` that constructs a dummy `HttpServletResponse` object so it overrides the method (along with the `@Deprecated` annotation).

**Comment by vladak** [comment]
Tomcat 10.1 uses [jakarta.servlet-api](https://search.maven.org/artifact/jakarta.servlet/jakarta.servlet-api) 6.0 while Tomcat 10.0 uses 5.0. The removal of deprecated APIs in jakarta.servlet-api happened via https://github.com/jakartaee/servlet/pull/419 (for 6.0.0).

**Comment by vladak** [comment]
For the record, with Tomcat 10.1, the webapp seems to work normally in terms of the UI. That is after after the fix for #4050.

**Comment by sindelar-fr** [comment]
For me this seems to also prevent the initial sync/indexing. I've tried the latest and master docker images, log from the master follows:
`Running version 1.7.36
synchronization period = 60 minutes
Deploying web application
Creating bare configuration in /opengrok/etc/configuration.xml
Oct 26, 2022 8:49:21 AM org.opengrok.indexer.index.Indexer parseOptions
INFO: Indexer options: [-s, /opengrok/src, -d, /opengrok/data, -c, /usr/local/bin/ctags, --remote, on, -H, -S, -W, /opengrok/etc/configuration.xml, --noIndex, -P]
Oct 26, 2022 8:49:21 AM org.opengrok.indexer.index.Indexer main
INFO: Indexer version 1.7.36 (aa5e7d1514dd07cd8985ac487a9b7e2b4edf146f) running on Java 11.0.16.1
Oct 26, 2022 8:49:21 AM org.opengrok.indexer.util.CtagsUtil validate
INFO: Using ctags: Universal Ctags 5.9.0(p5.9.20221023.0), Copyright (C) 2015-2022 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Oct 25 2022, 19:14:10
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +packcc, +optscript
Oct 26, 2022 8:49:21 AM org.opengrok.indexer.index.Indexer prepareIndexer
INFO: Scanning for repositories in [ removed :)), but be sure there's more than one ] (down to 2 levels below source root)
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Oct 26, 2022 8:49:25 AM org.opengrok.indexer.util.Statistics logIt
INFO: Done scanning for repositories, found 10 repositories (took 3.585 seconds)
Oct 26, 2022 8:49:25 AM org.opengrok.indexer.util.Statistics logIt
INFO: Done invalidating repositories (10 valid, 10 working) (took 94 ms)
Oct 26, 2022 8:49:25 AM org.opengrok.indexer.index.Indexer writeConfigToFile
INFO: Writing configuration to /opengrok/etc/configuration.xml
Oct 26, 2022 8:49:25 AM org.opengrok.indexer.index.Indexer writeConfigToFile
INFO: Done writing configuration to /opengrok/etc/configuration.xml
Oct 26, 2022 8:49:25 AM org.opengrok.indexer.util.Statistics logIt
INFO: Indexer finished (took 4.335 seconds)
Number of sync workers: 4
Waiting for Tomcat to start
Starting REST app on port 5000
Starting Tomcat
NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
26-Oct-2022 08:49:28.412 WARNING [main] org.apache.tomcat.util.digester.Digester.endElement No rules found matching [Server/Service/Engine/Host/Disabled]
26-Oct-2022 08:49:28.418 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version name:   Apache Tomcat/10.1.1
26-Oct-2022 08:49:28.418 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Oct 3 2022 12:42:14 UTC
26-Oct-2022 08:49:28.419 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version number: 10.1.1.0
26-Oct-2022 08:49:28.419 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
26-Oct-2022 08:49:28.419 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            5.4.0-131-generic
26-Oct-2022 08:49:28.420 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
26-Oct-2022 08:49:28.420 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /opt/java/openjdk
26-Oct-2022 08:49:28.420 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           11.0.16.1+1
26-Oct-2022 08:49:28.420 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Eclipse Adoptium
26-Oct-2022 08:49:28.420 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /usr/local/tomcat
26-Oct-2022 08:49:28.421 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /usr/local/tomcat
26-Oct-2022 08:49:28.433 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.lang=ALL-UNNAMED
26-Oct-2022 08:49:28.433 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.io=ALL-UNNAMED
26-Oct-2022 08:49:28.433 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.util=ALL-UNNAMED
26-Oct-2022 08:49:28.434 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
26-Oct-2022 08:49:28.434 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
26-Oct-2022 08:49:28.434 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/usr/local/tomcat/conf/logging.properties
26-Oct-2022 08:49:28.434 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
26-Oct-2022 08:49:28.434 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
26-Oct-2022 08:49:28.434 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
26-Oct-2022 08:49:28.435 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
26-Oct-2022 08:49:28.435 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.base=/usr/local/tomcat
26-Oct-2022 08:49:28.435 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.home=/usr/local/tomcat
26-Oct-2022 08:49:28.435 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.io.tmpdir=/usr/local/tomcat/temp
26-Oct-2022 08:49:28.439 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent Loaded Apache Tomcat Native library [2.0.1] using APR version [1.7.0].
26-Oct-2022 08:49:28.444 INFO [main] org.apache.catalina.core.AprLifecycleListener.initializeSSL OpenSSL successfully initialized [OpenSSL 3.0.2 15 Mar 2022]
26-Oct-2022 08:49:28.787 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8080"]
26-Oct-2022 08:49:28.818 INFO [main] org.apache.catalina.startup.Catalina.load Server initialization in [655] milliseconds
26-Oct-2022 08:49:28.874 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
26-Oct-2022 08:49:28.874 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet engine: [Apache Tomcat/10.1.1]
26-Oct-2022 08:49:28.913 INFO [main] org.apache.catalina.startup.HostConfig.deployWAR Deploying web application archive [/usr/local/tomcat/webapps/ROOT.war]
26-Oct-2022 08:49:28.930 WARNING [main] org.apache.tomcat.util.digester.SetPropertiesRule.begin Match [Context] failed to set property [antiJARLocking] to [true]
26-Oct-2022 08:49:32.943 INFO [main] org.apache.jasper.servlet.TldScanner.scanJars At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
26-Oct-2022 08:49:33.103 INFO [main] org.opengrok.indexer.Metrics.<clinit> configuring PrometheusRegistry
26-Oct-2022 08:49:33.215 INFO [main] org.opengrok.web.WebappListener.contextInitialized Starting webapp with version 1.7.36 (aa5e7d1514dd07cd8985ac487a9b7e2b4edf146f)
26-Oct-2022 08:49:33.217 INFO [main] org.opengrok.indexer.configuration.Configuration.read Reading configuration from /opengrok/etc/configuration.xml
26-Oct-2022 08:49:33.818 INFO [main] org.opengrok.indexer.util.Statistics.logIt Done invalidating repositories (10 valid, 10 working) (took 519 ms)
26-Oct-2022 08:49:33.847 WARNING [main] org.opengrok.indexer.framework.PluginFramework.reload Plugin directory not found or not readable: /opengrok/data/../plugins. All requests allowed.
26-Oct-2022 08:49:34.325 INFO [Thread-10] org.opengrok.suggest.Suggester.init No index directories found, exiting...
26-Oct-2022 08:49:34.360 INFO [Thread-10] org.opengrok.web.api.v1.suggester.provider.service.impl.SuggesterServiceImpl.scheduleRebuild Scheduling suggester rebuild in PT15H10M25.669983S
26-Oct-2022 08:49:35.277 INFO [main] org.hibernate.validator.internal.util.Version.<clinit> HV000001: Hibernate Validator 7.0.0.Final
26-Oct-2022 08:49:38.058 INFO [main] org.apache.catalina.startup.HostConfig.deployWAR Deployment of web application archive [/usr/local/tomcat/webapps/ROOT.war] has finished in [9,143] ms
26-Oct-2022 08:49:38.060 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/source]
26-Oct-2022 08:49:38.080 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/source] has finished in [19] ms
26-Oct-2022 08:49:38.086 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
26-Oct-2022 08:49:38.108 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [9290] milliseconds
Tomcat is ready
26-Oct-2022 08:49:39.514 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [org.opengrok.web.api.v1.RestApp] in context with path [] threw exception [org.glassfish.jersey.server.ContainerException: java.lang.NoSuchMethodError: 'void jakarta.servlet.http.HttpServletResponse.setStatus(int, java.lang.String)'] with root cause
        java.lang.NoSuchMethodError: 'void jakarta.servlet.http.HttpServletResponse.setStatus(int, java.lang.String)'
                at org.glassfish.jersey.servlet.internal.ResponseWriter.writeResponseStatusAndHeaders(ResponseWriter.java:147)
                at org.glassfish.jersey.server.ServerRuntime$Responder$1.getOutputStream(ServerRuntime.java:625)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.commitStream(CommittingOutputStream.java:171)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.flushBuffer(CommittingOutputStream.java:276)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.commit(CommittingOutputStream.java:232)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.close(CommittingOutputStream.java:247)
                at org.glassfish.jersey.message.internal.OutboundMessageContext.close(OutboundMessageContext.java:842)
                at org.glassfish.jersey.server.ContainerResponse.close(ContainerResponse.java:389)
                at org.glassfish.jersey.server.ServerRuntime$Responder.writeResponse(ServerRuntime.java:707)
                at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:373)
                at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:363)
                at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:258)
                at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
                at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
                at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
                at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
                at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
                at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
                at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
                at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
                at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
                at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
                at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:366)
                at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:319)
                at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.opengrok.web.CookieFilter.doFilter(CookieFilter.java:50)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.opengrok.web.AuthorizationFilter.doFilter(AuthorizationFilter.java:68)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.opengrok.web.StatisticsFilter.doFilter(StatisticsFilter.java:67)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
                at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
                at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
                at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:119)
                at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
                at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
                at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357)
                at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:400)
                at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
                at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:861)
                at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1739)
                at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
                at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
                at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
                at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
                at java.base/java.lang.Thread.run(Thread.java:829)
could not list projects from web application: 500 Server Error:  for url: http://localhost:8080/api/v1/projects
26-Oct-2022 08:49:39.544 SEVERE [http-nio-8080-exec-4] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [org.opengrok.web.api.v1.RestApp] in context with path [] threw exception [org.glassfish.jersey.server.ContainerException: java.lang.NoSuchMethodError: 'void jakarta.servlet.http.HttpServletResponse.setStatus(int, java.lang.String)'] with root cause
        java.lang.NoSuchMethodError: 'void jakarta.servlet.http.HttpServletResponse.setStatus(int, java.lang.String)'
                at org.glassfish.jersey.servlet.internal.ResponseWriter.writeResponseStatusAndHeaders(ResponseWriter.java:147)
                at org.glassfish.jersey.server.ServerRuntime$Responder$1.getOutputStream(ServerRuntime.java:625)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.commitStream(CommittingOutputStream.java:171)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.flushBuffer(CommittingOutputStream.java:276)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.commit(CommittingOutputStream.java:232)
                at org.glassfish.jersey.message.internal.CommittingOutputStream.close(CommittingOutputStream.java:247)
                at org.glassfish.jersey.message.internal.OutboundMessageContext.close(OutboundMessageContext.java:842)
                at org.glassfish.jersey.server.ContainerResponse.close(ContainerResponse.java:389)
                at org.glassfish.jersey.server.ServerRuntime$Responder.writeResponse(ServerRuntime.java:707)
                at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:373)
                at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:363)
                at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:258)
                at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
                at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
                at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
                at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
                at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
                at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
                at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
                at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
                at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
                at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
                at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:366)
                at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:319)
                at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.opengrok.web.CookieFilter.doFilter(CookieFilter.java:50)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.opengrok.web.AuthorizationFilter.doFilter(AuthorizationFilter.java:68)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.opengrok.web.StatisticsFilter.doFilter(StatisticsFilter.java:67)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158)
                at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
                at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
                at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
                at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:119)
                at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
                at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
                at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357)
                at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:400)
                at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
                at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:861)
                at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1739)
                at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
                at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
                at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
                at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
                at java.base/java.lang.Thread.run(Thread.java:829)
Waiting for reindex to be triggered
could not list projects from web application: 500 Server Error:  for url: http://localhost:8080/api/v1/projects`
1.7.35 seems to be working more or less fine and when I perform the initial index with it, the latest/master seems to be able to display what's already indexed.

**Comment by vladak** [comment]
> For me this seems to also prevent the initial sync/indexing. I've tried the latest and master docker images, log from the master follows:

Thanks for the report, will constrain Tomcat to 10.0 for the Docker images for now. Also, the failures make sense as the indexer and related tooling rely on the RESTful APIs a lot.

**Comment by vladak** [comment]
I have a feature branch with some of the necessary changes, however it currently suffers from something similar to https://github.com/eclipse-ee4j/jersey/issues/4617

**Comment by vladak** [comment]
> Also, Jersey does not seem to be ready for the new servlet-api ([eclipse-ee4j/jersey#5181](https://github.com/eclipse-ee4j/jersey/issues/5181)) so this is on hold.

Jersey 3.1.0 was released so this part is unblocked.
